### ISSUE TITLE
manual LR model with structure proposed in paper

### ISSUE THREAD

**Issue by atkinssamuel** [atkinssamuel/eto#3]
need to do the following:

1. derive the equations behind the model
2. implement the equations and ensure strong results

**Comment by atkinssamuel** [comment]
https://eprint.iacr.org/2018/662.pdf

**Comment by atkinssamuel** [comment]
So far, the following have been implemented:

- Mini-batch GD
- NAG optimizer

To be implemented:
- Taylor approx of sigmoid function for the activation function
- Taylor approx of the sigmoid function for the gradient of the loss

**Comment by atkinssamuel** [comment]
I have now implemented the various approximations of the sigmoid activation function. The problem now is the following:

The outputs using the approximations of the sigmoid activation functions do not range from 0 to 1. They range from -10000 to 10000. This is problematic because the negative log-likelihood function is expecting values in the 0 to 1 range. It may be possible to clamp the range of these numbers by changing the weight initialization function, by rescaling the output using a scaling factor, or by changing the dataset to a dataset with smaller feature values. 

To summarize, the following solutions will be evaluated:

1. New weight initialization functions will be tested
2. A rescaling factor will be attempted
3. A new dataset will be tested

**Comment by atkinssamuel** [comment]
Update:

- Zero init was tested and did not help
- A rescaling factor was implemented and caused the model to only predict one class
- A regressional dataset was implemented using sklearn's make regression
  - The library is now able to solve regression problems as well
